bpdm-cleaning-service-dummy/src/test/kotlin/org/eclipse/tractusx/bpdm/cleaning/service/CleaningServiceDummyTest.kt

```diff
@@ -0,0 +1,185 @@
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.cleaning.service
+
+import org.eclipse.tractusx.bpdm.cleaning.testdata.CommonValues.businessPartnerWithBpnA
+import org.eclipse.tractusx.bpdm.cleaning.testdata.CommonValues.businessPartnerWithBpnLAndBpnAAndLegalAddressType
+import org.eclipse.tractusx.bpdm.cleaning.testdata.CommonValues.businessPartnerWithBpnSAndBpnAAndLegalAndSiteMainAddressType
+
+import org.eclipse.tractusx.bpdm.cleaning.testdata.CommonValues.businessPartnerWithEmptyBpnAndSiteMainAddressType
+import org.eclipse.tractusx.bpdm.cleaning.testdata.CommonValues.businessPartnerWithEmptyBpnLAndAdditionalAddressType
+import org.eclipse.tractusx.bpdm.common.dto.*
+import org.eclipse.tractusx.orchestrator.api.model.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import java.time.Instant
+import java.util.*
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+class CleaningServiceDummyTest @Autowired constructor(
+    val cleaningServiceDummy: CleaningServiceDummy,
+) {
+
+
+    @Test
+    fun `test processCleaningTask with BpnA present and additional address type`() {
+        val taskStepReservationEntryDto = createSampleTaskStepReservationResponse(businessPartnerWithBpnA).reservedTasks[0]
+
+        val result = cleaningServiceDummy.processCleaningTask(taskStepReservationEntryDto)
+
+        val expectedBpnA = taskStepReservationEntryDto.businessPartner.generic.bpnA
+
+        val resultedAddress = result.businessPartner?.address
+
+        val resultedLegalEntity = result.businessPartner?.legalEntity
+
+        // legalEntity should be Generated with new bpnL and legalAddress bpnA
+        // addressPartner should use passed bpnA, and it will be different from legalAddress since type is additional address type
+
+        assertEquals(expectedBpnA, resultedAddress?.bpnAReference?.referenceValue)
+
+        assertEquals(BpnReferenceType.BpnRequestIdentifier, resultedLegalEntity?.bpnLReference?.referenceType)
```

Comment:
You only test whether the references are as expected. I would expect tests which also check whether the expected full business partner values are created as well

Reply:
For this part already made changes so it can have the all body checked and api calls